zerver/tests/test_auth_backends.py

```diff
@@ -1891,6 +1891,65 @@ def test_validate_state(self) -> None:
             self.assertIn('login', result.url)
             m.assert_called_with("Sign in with Apple failed: bad state token.")
 
+class AppleAuthBackendNativeFlowTest(AppleIdAuthBackendTest):
+    SIGNUP_URL = '/complete/apple/'
```

Comment:
I think we need to do LOGIN_URL as well?

Reply:
I'm a bit confused about why this is even working without overriding LOGIN_URL

Reply:
It works because we set ` alternative_start_url=self.AUTH_FINISH_URL` in `prepare_login_url_and_headers` call.
we cannot use it for signup case because even if we give an `alternative_start_url` it's gets  changed to `SIGNUP_URL`. (I guess it makes sense to have it like that for other backends)

Reply:
I think we do not need to change `LOGIN_URL` and we will not be having a `LOGIN_URL` for native test class as we do not include that in the mixin. The tests should work perfectly as we do not go to `LOGIN_URL` at any point in native flow afaik.

Reply:
There's this test `test_social_auth_session_fields_cleared_correctly` that uses actual `prepare_login_url_and_headers` which uses actual `LOGIN_URL` and doesn't make use of `social_auth_test`. 
If we're using the actual `LOGIN_URL` there's no difference with running this test in `AppleIdAuthBackendTest` and `AppleAuthBackendNativeFlowTest`.

I added a test overriding this doing it right way for this flow.